math/src/main/java/com/powsybl/math/matrix/SparseLUDecomposition.java

```diff
@@ -32,10 +36,22 @@
 
     private native void release(String id);
 
+    private native void update(String id, int[] ap, int[] ai, double[] ax);
+
     private native void solve(String id, double[] b);
 
     private native void solve2(String id, int m, int n, ByteBuffer b);
 
+    /**
+     * {@inheritDoc}
+     *
+     * The structure of the matrix is not supposed to have changed, only non zero values.
```

Comment:
Do you know what's the behaviour of the structure has been changed by mistake ?

Reply:
> No blocking issue, only minor comments about the doc.
> 
> Still, it has not much to do with the current evolution but I am not comfortable with the this aspect of the design: SparseMatrix is actually a restriction of the general Matrix API, which makes it easy to write code which does not work for sparse implementation. It does not respect the so-called "LSP" considered as good practice.
> 
> The implementation of some method are very specific for a target use : for instance the natural implementation of the `reset` method would be to completely clear the array of elements, but it does not answer our need to only reset and update the elements value.
> 
> I have nothing better to propose at this point though :)

You are right about the overall design weakness. I think it is somehow related to #726 as matrix mutation methods should be move from Matrix to something like builder/filler class and Matrix should be immutable. I add your comment the issue.

Reply:
> Do you know what's the behaviour of the structure has been changed by mistake ?

Not tested and not described in KLU doc.

Reply:
To robustify the code, I can save the element count in SparseLuDecomposition after each decomposition and throw an exception if element count has increased since last decomposition?

Reply:
Why not ! Maybe the best check would be to compare all arrays except values, but maybe it's too expensive ?

Reply:
Ah but it's impossible anyway except if we keep them in memory ... wrong idea :)

Reply:
sparse impl only allow adding new elements in the right order, I think just checking values.size() is enough to ensure no new element have been added

Reply:
New change seems to solve this issue?